File: app/src/main/java/com/innso/mobile/ui/fragments/CompanyFragment.java
package com.innso.mobile.ui.fragments;

import androidx.databinding.DataBindingUtil;
import android.graphics.Typeface;
import android.os.Bundle;
import androidx.annotation.Nullable;
import android.text.style.RelativeSizeSpan;
import android.text.style.StyleSpan;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.innso.mobile.R;
import com.innso.mobile.databinding.FragmentCompanyBinding;
import com.innso.mobile.ui.BaseFragment;
import com.innso.mobile.ui.viewModels.CompanyViewModel;
import com.innso.mobile.utils.StringUtils;

public class CompanyFragment extends BaseFragment {

    private CompanyViewModel viewModel;

    private FragmentCompanyBinding binding;

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = DataBindingUtil.inflate(inflater, R.layout.fragment_company, container, false);
        viewModel = new CompanyViewModel();
        binding.setViewModel(viewModel);
        return binding.getRoot();
    }

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        initView();
    }

    @Override
    public void onResume() {
        super.onResume();
        subscribe();
    }

    private void subscribe(){
        disposable.add(viewModel.startActivityEvent().subscribe(this::startActivity));
    }

    private void initView() {
        binding.textViewCompanyName.setText(StringUtils.setSpannablesFromRegex(getString(R.string.company_name),
                "^(.*?)\\n", new StyleSpan(Typeface.BOLD), new RelativeSizeSpan(1.2f)));
        binding.textViewCompanyNit.setText(StringUtils.setSpannablesFromRegex(getString(R.string.company_nit), "(.*):", new StyleSpan(Typeface.BOLD)));
    }

}

File: app/src/main/java/com/innso/mobile/ui/fragments/CompanyFragment.kt
package com.innso.mobile.ui.fragments

import android.graphics.Typeface
import android.os.Bundle
import android.text.style.RelativeSizeSpan
import android.text.style.StyleSpan
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.databinding.DataBindingUtil
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProviders
import com.innso.mobile.R
import com.innso.mobile.databinding.FragmentCompanyBinding
import com.innso.mobile.ui.BaseFragment
import com.innso.mobile.utils.StringUtils
import com.innso.mobile.viewModels.company.CompanyViewModel

class CompanyFragment : BaseFragment() {

    private lateinit var viewModel: CompanyViewModel

    private lateinit var binding: FragmentCompanyBinding

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        super.onCreate(savedInstanceState)
        binding = DataBindingUtil.inflate(inflater, R.layout.fragment_company, container, false)
        initViewModel()
        binding.viewModel = viewModel
        return binding.root
    }

    fun initViewModel() {
        binding.lifecycleOwner = this
        viewModel = ViewModelProviders.of(this).get(CompanyViewModel::class.java)
        viewModel.startActivity().observe(this, Observer { startActivity(it) })
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        initView()
    }

    private fun initView() {
        binding.textViewCompanyName.text = StringUtils.setSpannablesFromRegex(getString(R.string.company_name),
                "^(.*?)\\n", StyleSpan(Typeface.BOLD), RelativeSizeSpan(1.2f))
        binding.textViewCompanyNit.text = StringUtils.setSpannablesFromRegex(getString(R.string.company_nit), "(.*):", StyleSpan(Typeface.BOLD))
    }
}
